lib/type.ts

```diff
@@ -45,9 +45,8 @@ class TypeTranspiler extends base.TranspilerBase {
         this.emit('>');
         break;
       case ts.SyntaxKind.FunctionType:
-        this.emit('dynamic /*');
-        this.emit(node.getText());
-        this.emit('*/');
+        this.reportError(node, 'Inline function type declarations not supported. ' +
```

Comment:
It does not support anonymous function type for local non-parameter variables (more on that below). However, every anonymous function type declaration can also be expressed via a private _named_ function type alias, so this is not a regression but merely a redirection to use another way of doing the same thing, one that ts2dart is able to translate.

Alternatives are:

- (status quo) allow anonymous function types but erase them during transpilation: this is bad because users will feel that they can use this while they shouldn't, otherwise Dart code quality will hurt
- generate a named function type: however this will require that we generate a name for it, which is unlikely to be meaningful. Again, Dart code quality will hurt.

However, Dart does support anonymous type aliases for parameters. I'll see if I can squeeze in another pull request for that. This what they look like:

```dart
func(void fn(int));
```